File: MidoClient/src/app/models/dtos/UserDto.ts
import { User } from "../User";

export class UserDto {
    id?: number;
    username: string;
    password: string;
    email: string;
    role: string;
  
    constructor(user: User) {
      this.id = user.id;
      this.username = user.username;
      this.password = user.password;
      this.email = user.email;
      this.role = user.role;
    }
}

File: src/main/java/com/mido/dtos/UserDto.java
package com.mido.dtos;

import com.mido.models.Role;
import com.mido.models.Status;
import lombok.Data;

@Data
public class UserDto {
    private String username;
    private String password;
    private String email;
    private Role role;
    private Status status;
}
